### ISSUE TITLE
Update GPS Validity Check to Use mode Instead of status Field

### ISSUE THREAD

**Issue by danthony06** [swri-robotics/gps_umd#50]
Rather than checking the ```status``` field for the fix validity, we should be looking at the ```mode``` field. These fields are described here: https://gpsd.gitlab.io/gpsd/gpsd_json.html. Currently, the checks on the ```status``` field will work correctly with basic GPS receivers, but may not work with RTK, DGPS, or other receivers, and may report that we do not have a fix, even if the receivers are working correctly. This change will bring us closer to how GPSD is intended to be used, and will also help resolve issues with the ```status``` values changing between gpsd releases that we're continuously putting ```ifdef``` statements in for.

This will become especially important around gpsd v3.23 gets released, as the status fields have significantly changed, and the conditionally compiling around them will be quite ugly. It will also allow us to remove many of the ```ifdef``` statements that are littering the code and making it challenging to read and maintain.

**Comment by flabrosse** [comment]
Yes please!.  At the moment, on ubuntu 24.04, ROS Jazzy, the status is always -1 with my reach module, even when the module says it has a (single) fix.  This seems to be blocking other things down the line (navsat_transform for example).

**Comment by engnfrc** [comment]
Where are we at with this issue? I don't understand why when I have a valid fix the code is setup for status.status = -1 (STATUS_NO_FIX), that's just wrong and as mentioned in previous comments causes problems across the system. Last updates are nearing a year old.

**Comment by danthony06** [comment]
I'll try to get to it this week.

**Comment by engnfrc** [comment]
@danthony06 please take a look at this PR: https://github.com/swri-robotics/gps_umd/pull/119

When I first stumbled on this issue I only half understood what was going on. My biggest issue was that I was seeing NavSatFix reporting a status of -1 when there's a 3D fix present without corrections. I've applied using the mode to initialize the status to either no fix or fix and then if there are SBAS, DGPS, RTK Fixed, or RTK Float corrections then I upgrade the GPSFix/NavSatFix status fields accordingly. I also set the service field to sane defaults.

I've also noticed that the used PRN values have been an array of 0 and 1, which ended up being a bug in the code and now the correct PRN values are applied.

I went ahead and had ChatGPT check over the file and it had a number of other suggestions that I've included. I tried to keep backwards compatibility in mind while making the changes.

**Comment by danthony06** [comment]
Big thanks to @engnfrc for taking care of this. I've just bloomed this fix that will be in v2.3.0 for `humble` and v3.0.0 for all other distros.
